modules/hal_afbr/platform_malloc.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/check.h>
+
+#define DT_DRV_COMPAT		brcm_afbr_s50
+#define NUM_AFBR_INST		DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)
+
+BUILD_ASSERT(NUM_AFBR_INST > 0, "Invalid number of AFBR-S50 instances");
+
+/** Defined separate memslab to isolate library from the other components.
+ * Through debugging, the library requests an initial allocation of ~4-KiB,
+ * which is why the total pool is sized to 8-KiB per instnace.
+ */
+K_MEM_SLAB_DEFINE(argus_memslab, 64, 128 * NUM_AFBR_INST, sizeof(void *));
+
+void *Argus_Malloc(size_t size)
+{
+	void *ptr = NULL;
+	int err;
+
+	err = k_mem_slab_alloc(&argus_memslab, &ptr, K_NO_WAIT);
+
+	CHECKIF(err != 0 || ptr == NULL) {
+		return NULL;
+	}
+
+	return ptr;
+}
+
+void Argus_Free(void *ptr)
+{
+	k_mem_slab_free(&argus_memslab, &ptr);
```

Comment:
```suggestion
	k_mem_slab_free(&argus_memslab, ptr);
```

Reply:
This is a nice catch. It is not impacting the code now because the sensor driver is not De-init'ing the instance. However, I can confirm that calling `Argus_DestroyHandle()` which calls `Argus_Free()` triggers an assert.

Side-note: This has bitten me before while migrating a production app (when this behavior changed in 3.5). So it's very much appreciated! :smile: 
https://github.com/zephyrproject-rtos/zephyr/blob/e462ef3c0747be6ea54bcae6fb292cbd49a0c505/doc/releases/migration-guide-3.5.rst?plain=1#L20-L26

I'll iterate addressing this issue.